pyccl/halos/hmfunc/jenkins01.py

```diff
@@ -11,33 +12,28 @@ class MassFuncJenkins01(MassFunc):
     This parametrization is only valid for 'fof' masses.
 
     Args:
-        cosmo (:class:`~pyccl.core.Cosmology`): A Cosmology object.
         mass_def (:class:`~pyccl.halos.massdef.MassDef`):
             a mass definition object.
             this parametrization accepts FoF masses only.
-            If `None`, FoF masses will be used.
+            The default is 'fof'.
         mass_def_strict (bool): if False, consistency of the mass
             definition will be ignored.
     """
     name = 'Jenkins01'
 
-    def __init__(self, cosmo, mass_def=None, mass_def_strict=True):
-        super(MassFuncJenkins01, self).__init__(cosmo,
-                                                mass_def=mass_def,
-                                                mass_def_strict=True)
+    @warn_api
+    def __init__(self, *,
+                 mass_def=MassDef('fof', 'matter'),
```

Comment:
I've only noted this once or twice above but, again, this code (the initialisation of the mass definition) gets run every time you import CCL now. Not just this, but all the instances of this in all the classes. Please don't do this and go back to default=None

Reply:
If you want to convince yourself, create a module with the following code, and then import it

```
class A:
    def __init__(self):
        print("This will get printed on import.")


def ay(a=A()):
    return
```

Reply:
Yes, it is run on import, as I mentioned above. And it's super fast, we shouldn't worry about that. Initializing a `MassDef` object takes 3 μs. The limit of human time perception is ca. 15 ms, so it would take thousands of those for anyone to realize any delay on import.

Reply:
There are only 2 cases where this is an anti-pattern (and potentially dangerous) in Python:
- when the default is mutable and is actually mutated in that function
- when there is a complicated dependency tree so you get into partially initialized module errors.

In all other cases, it is actually encouraged to do it this way because you provide clearer code and users can see the default when they run with IPython or similar.

Reply:
As I said above, you need to revert all of these either to None, or default strings. I don't want function calls in default arguments.

Reply:
Like I said, this is considered to be pretty standard in any Python library. In our particular case, the extra time is negligible. For comparison, it takes 60 μs in total to pre-initialize those in the entire library, while it takes 1 ms (15x more) to set the functions with `cosmo` as their first argument to be methods of `Cosmology`.  So I don't see how this is jeopardizing efficiency at all.

Either way, what you are proposing with the strings, is the single thing that makes more sense than pre-initializing them. I extended the functionality of the halo model ingredients to accept a name string too, and initialize it internally, which is even more simple for the user.